src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +35,113 @@
+            predicate = new PhoneContainsKeywordsPredicate(trimmedPhone);
+        } else if (targetPrefix.equals(PREFIX_EMAIL)) {
+            List<String> email = argMultimap.getAllValues(PREFIX_EMAIL);
+            List<String> trimmedEmail = prepareArguments(email);
+            predicate = new EmailContainsKeywordsPredicate(trimmedEmail);
+        } else if (targetPrefix.equals(PREFIX_ADDRESS)) {
+            List<String> address = argMultimap.getAllValues(PREFIX_ADDRESS);
+            List<String> trimmedAddress = prepareArguments(address);
+            predicate = new AddressContainsKeywordsPredicate(trimmedAddress);
+        } else if (targetPrefix.equals(PREFIX_TAG)) {
+            List<String> tagList = argMultimap.getAllValues(PREFIX_TAG);
+            List<String> trimmedTagList = prepareArguments(tagList);
+            predicate = new TagContainsKeywordsPredicate(trimmedTagList);
+        } else {
+            throw new ParseException("Something bad has happened during parsing. Please contact the project team!");
+        }
+        return new FindCommand(predicate);
+    }
+
+    /**
+     * Returns a list of prefixes that can be found in the arguments list.
+     * @param args
+     */
+    private static List<Prefix> getPrefixes(String args) {
+        HashMap<String, Prefix> prefixMapping = new HashMap<>();
+        prefixMapping.put("n/", PREFIX_NAME);
+        prefixMapping.put("a/", PREFIX_ADDRESS);
+        prefixMapping.put("e/", PREFIX_EMAIL);
+        prefixMapping.put("p/", PREFIX_PHONE);
+        prefixMapping.put("t/", PREFIX_TAG);
+        List<Prefix> prefixesInList = new ArrayList<>();
+
+        for (Map.Entry<String, Prefix> entry : prefixMapping.entrySet()) {
+            if (args.contains(entry.getKey())) {
+                prefixesInList.add(entry.getValue());
+            }
+        }
+        return prefixesInList;
+    }
+
+    /**
+     * Prepares the argument list to be searched by ensuring that each argument is a single word without
+     * any leading or ending whitespaces.
+     * @param potentialArgumentsList
+     */
+    private static List<String> prepareArguments(List<String> potentialArgumentsList) {
+        List<String> preparedArgumentsList = new ArrayList<>();
+        for (String arg : potentialArgumentsList) {
+            String[] element = arg.split(" ");
+            for (String subElement : element) {
+                String trimmedSubElement = subElement.trim();
+                if (!trimmedSubElement.isEmpty()) {
+                    preparedArgumentsList.add(trimmedSubElement);
+                }
+            }
+        }
+        return preparedArgumentsList;
+    }
+
+    /**
+     * Returns true if the target prefix contains at least one non-empty {@code Optional} value in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean isAPrefixWithValue(ArgumentMultimap argumentMultimap, Prefix prefix) {
```

Comment:
Possible duplicate of `AddCommandParser.arePrefixesPresent`? The implementation seems more concise also.

Reply:
I saw that implementation before when I was first implementing this. 

The difference between that particular implementation and my own implementation is that my implementation will check for the case where a prefix does not have a value (e.g. `find n/`) and throw an `INVALID_COMMAND_FORMAT` message but the given implementation will not. This is because for `AddCommandParser`, this issue is caught by the `parse` method in `ParserUtil`, but I did not use this particular method because I only need to manipulate the string form of `name`/`email`/`phone`/`address`/`tag` and do not need to instantiate an instance of the corresponding object for the `find` command.